spec/bolt/transport/ssh/connection_spec.rb

```diff
@@ -0,0 +1,128 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+require 'net/ssh'
+require 'bolt/inventory'
+require 'bolt/transport/ssh/connection'
+require 'bolt_spec/errors'
+
+describe Bolt::Transport::SSH::Connection do
+  include BoltSpec::Errors
+
+  let(:uri) { 'ssh://foo.example.com' }
+  let(:inventory) { Bolt::Inventory.empty }
+  let(:target) { inventory.get_target(uri) }
+  let(:transport_logger) { Logging.logger[Net::SSH] }
+  let(:subject) { described_class.new(target, transport_logger) }
+
+  context "when setting user" do
+    before :each do
+      allow(Net::SSH::Config).to receive(:for).and_return(user: 'sshuser')
+      allow(Etc).to receive(:getlogin).and_return('loginuser')
+    end
+
+    it "uses the target's user if one is set" do
+      inventory.set_config(target, 'ssh', 'user' => 'targetuser')
+      expect(subject.user).to eq('targetuser')
+    end
+
+    it "uses the SSH config user if no target user is set" do
+      expect(subject.user).to eq('sshuser')
+    end
+
+    it "falls back to the login user if no SSH config or target user is set" do
+      allow(Net::SSH::Config).to receive(:for).and_return({})
+      expect(subject.user).to eq('loginuser')
+    end
+
+    it "doesn't check SSH config if load-config is false" do
+      inventory.set_config(target, 'ssh', 'load-config' => false)
+      expect(subject.user).to eq('loginuser')
+    end
+  end
+
+  context "when connecting", ssh: true do
+    it "passes proxyjump options" do
+      inventory.set_config(target, 'ssh', 'proxyjump' => 'jump.example.com')
+
+      allow(Net::SSH).to receive(:start) do |_, _, options|
+        expect(options[:proxy]).to be_instance_of(Net::SSH::Proxy::Jump)
+      end
```

Comment:
I think this needs to return a stubbed object so `transport.server_version.version` succeeds